iron-image.html

```diff
@@ -298,65 +330,43 @@
         this.style.height = isNaN(this.height) ? this.height : this.height + 'px';
       },
 
+      _preventLoadChanged: function() {
+        if (this.preventLoad || this.loaded) return;
+
+        this._reset();
+        this._load(this.src);
+      },
+
       _srcChanged: function(newSrc, oldSrc) {
-        if (newSrc !== oldSrc) {
-          this.loaded = false;
+        if (newSrc === oldSrc) return;
```

Comment:
Added, also I changed `_matchesImgSrc` to `_resolveSrc` (which doesn't actually test, just resolves) so that it doesn't reset multiple times when `preventLoad` is true.